src/process/personality.rs

```diff
@@ -0,0 +1,1747 @@
+use core::convert::{TryFrom, TryInto};
+use core::num::NonZeroU64;
+use core::ptr::NonNull;
+use core::sync::atomic::AtomicU8;
+use core::{mem, ptr};
+
+use bitflags::bitflags;
+
+use crate::fd::{AsRawFd, BorrowedFd};
+use crate::ffi::CString;
+use crate::process::{Pid, RawNonZeroPid, RawPid};
+use crate::{backend, io};
+
+use backend::c::{c_int, c_uint, c_ulong};
+use backend::process::syscalls::prctl as syscall_prctl;
+pub use backend::process::types::Signal;
+
+/// `prctl()`—operations on a process or thread.
+///
+/// The type of `arg2` to `arg5` is [`usize`], instead of the most accurate
+/// type [`c_ulong`]. This is due to the assumption that the target platform
+/// ABI will always be of type `*LP32` or `*LP64` where the following is true:
+/// ```
+/// assert_eq!(std::mem::size_of::<std::os::raw::c_ulong>(), std::mem::size_of::<usize>());
+/// ```
+///
+/// # Safety
+///
+/// This function should be declared `unsafe` because, for some values of `options`,
+/// some of `arg2` to `arg5` are interpreted as pointers that are dereferenced by the kernel.
+///
+/// # References
+///  - [Linux]
+///
+/// [Linux]: https://man7.org/linux/man-pages/man2/prctl.2.html
+#[inline]
+pub fn prctl(option: i32, arg2: usize, arg3: usize, arg4: usize, arg5: usize) -> io::Result<i32> {
```

Comment:
Is there a reason this function isn't actually declared `unsafe`?

Also, is there a need for this function to be `pub`? Even though it's `unsafe`, it's a goal of Rustix to expose typed APIs, rather than just blanket APIs like `ioctl` or `fcntl`. It's a tradeoff, and I'm open to different approaches. But if there isn't a need for a fully general `prctl` in the public API, I'd prefer to keep it private.

Reply:
I'll remove `prctl()` altogether. I assume users who want to call the original version would use another crate.